src/read/archive.rs

```diff
@@ -44,10 +46,48 @@ impl<'data, R: ReadRef<'data>> ArchiveFile<'data, R> {
         let magic = data
             .read_bytes(&mut tail, archive::MAGIC.len() as u64)
             .read_error("Invalid archive size")?;
-        if magic != &archive::MAGIC[..] {
+        let mut kind_by_header = ArchiveKind::Unknown;
+        if magic == &archive::AIX_BIG_MAGIC[..] {
+            kind_by_header = ArchiveKind::AixBig;
+        } else if magic != &archive::MAGIC[..] {
             return Err(Error("Unsupported archive identifier"));
         }
 
+        if kind_by_header == ArchiveKind::AixBig {
```

Comment:
```suggestion
        if magic == &archive::AIX_BIG_MAGIC {
```
There's no need for the `kind_by_header` variable. Do the `AIX_BIG_MAGIC` check here, and put this block before the existing `archive::MAGIC` check.